modules/gdscript/gdscript_parser.cpp

```diff
@@ -8152,6 +8156,11 @@ void GDScriptParser::_check_block_types(BlockNode *p_block) {
 							break;
 						}
 
+						if (op->arguments[0]->type == Node::TYPE_SELF) {
```

Comment:
I don't think this second check is needed. It'll have already given an error in the first one.

Reply:
I assumed it'll be necessary since it's checked for constants in both locations, is the constant check also redundant?

Reply:
No, because the constant check here is for typed stuff, which aren't understood by the tokenizer (so it catches `const`s defined on script, or engine-defined such as `Node`). 

In fact, we could also make `self` as constant by adding `node_type.is_constant = true;` to: https://github.com/godotengine/godot/blob/ff7e7bd260f8541d4c3303a8c82ae438402e71c3/modules/gdscript/gdscript_parser.cpp#L6289-L6293

Which would be enough to disallow reassign to `self`. Though I prefer your change anyway since the error message is clearer. Feel free to include this suggestion in the commit. But this second check is still redundant.